test-d/index.test-d.ts

```diff
@@ -81,3 +86,25 @@ memoize((_arguments: {key: string}) => 1, {
 		clear: () => undefined,
 	},
 });
+
+// Test that exported CacheItem and CacheLike types can be used to define custom caches
+type Key = string;
+type Value = number;
+
+// Test CacheItem type
+const cacheItem: CacheItem<Value> = {
+	data: 42,
+	maxAge: Date.now() + 1000,
+};
+expectType<Value>(cacheItem.data);
+expectType<number>(cacheItem.maxAge);
+
+// Test CacheLike type with Map-based implementation
+const customCache: CacheLike<Key, Value> = new Map<Key, CacheItem<Value>>();
+expectType<boolean>(customCache.has('test'));
+expectType<CacheItem<Value> | undefined>(customCache.get('test'));
+
+// Test using CacheLike in memoize options
+const customCacheMap: CacheLike<string, number> = new Map();
+const memoizedWithCustomCache = memoize(Number, {cache: customCacheMap});
+expectType<typeof Number>(memoizedWithCustomCache);
```

Comment:
These changes are not needed. We don't need tests to confirm that, yes, the two added `export` keywords do indeed work.